f5/bigip/tm/ltm/profile.py

```diff
@@ -56,9 +108,1070 @@ def __init__(self, client_ssls):
         self._meta_data['required_json_kind'] =\
             'tm:ltm:profile:client-ssl:client-sslstate'
 
-    def create(self, certname, keyname):
-        payload = {"name": certname[:-4],
-                   "cert": certname,
-                   "key": keyname}
-        logging.debug(payload)
-        return self._create(**payload)
+    def create(self, **kwargs):
+        """Allows creation of SSL profile with just key/cert names.
+
+            Will require 'name' kwarg to be provided otherwise
+
+        """
+        if 'certname' and 'keyname' in kwargs:
```

Comment:
Let's change the structure here:

`certname`, `keyname`, and `name` are now required arguments.

So let's put something like this in place:

https://gist.github.com/zancas/aab07faf9e55b03e027fc111f1f71d6d